frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs

```diff
@@ -225,6 +230,81 @@
         (when (or editing? creating?)
           [:div {:class (stl/css  :counter)} (str size "/300")])]])))
 
+
+(mf/defc component-variant*
+  {::mf/props :obj}
+  [{:keys [component shape libraries]}]
+  (let [id-component (:id component)
+        properties   (:variant-properties component)
+        data         (get-in libraries [(:component-file shape) :data])
```

Comment:
Not an issue, just a note: we probably need to revisit this when feature is ready, because passing all around the whole libraries map has very bad implications on performance.

Reply:
So i'm not going to comment  more performance issues, because the whole component menu design is pretty bad with performance, so this should be addressed later, once the feature is complete or at least the sidebar menu part ready.